pinot-common/src/main/java/org/apache/pinot/common/audit/AuditRequestProcessor.java

```diff
@@ -140,10 +165,15 @@ private AuditEvent.AuditRequestPayload captureRequestPayload(ContainerRequestCon
       }
 
       final AuditConfig config = _configManager.getCurrentConfig();
-      if (config.isCaptureRequestHeaders()) {
-        MultivaluedMap<String, String> headers = requestContext.getHeaders();
-        if (!headers.isEmpty()) {
-          payload.setHeaders(toMap(headers));
+
+      Set<String> allowedHeaders = parseAllowedHeaders(config.getCaptureRequestHeaders());
+      if (!allowedHeaders.isEmpty()) {
+        MultivaluedMap<String, String> allHeaders = requestContext.getHeaders();
+        if (!allHeaders.isEmpty()) {
+          Map<String, Object> filteredHeaders = toMap(allHeaders, allowedHeaders);
+          if (!filteredHeaders.isEmpty()) {
+            payload.setHeaders(filteredHeaders);
+          }
```

Comment:
is there a use case / scenario where we may want to capture all headers? should wildcard `*` be allowed?